src/databricks/sql/auth/endpoint.py

```diff
@@ -0,0 +1,107 @@
+#
+# It implements all the cloud specific OAuth configuration/metadata
+#
+#   Azure: It uses AAD
+#   AWS: It uses Databricks internal IdP
+#   GCP: Not support yet
+#
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Optional, List
+import os
+
+OIDC_REDIRECTOR_PATH = "oidc"
+
+
+class OAuthScope:
+    OFFLINE_ACCESS = "offline_access"
+    SQL = "sql"
+
+
+class CloudType(Enum):
+    AWS = "aws"
+    AZURE = "azure"
+    GCP = "gcp"
+
+
+# Infer cloud type from Databricks SQL instance hostname
+def infer_cloud_from_host(hostname: str) -> Optional[CloudType]:
+    # normalize
+    host = hostname.lower().replace("https://", "").split("/")[0]
+
+    if host.endswith(".azuredatabricks.net"):
+        return CloudType.AZURE
+    elif host.endswith(".gcp.databricks.com"):
+        return CloudType.GCP
+    elif host.endswith("cloud.databricks.com"):
+        return CloudType.AWS
+    else:
+        return None
+
+
+def get_databricks_oidc_url(hostname: str):
+    maybe_scheme = "https://" if not hostname.startswith("https://") else ""
+    maybe_trailing_slash = "/" if not hostname.endswith("/") else ""
+    return f"{maybe_scheme}{hostname}{maybe_trailing_slash}{OIDC_REDIRECTOR_PATH}"
+
+
+class OAuthEndpoints(ABC):
+    @abstractmethod
+    def get_scopes_mapping(self, scopes: List[str]) -> List[str]:
```

Comment:
Is there anywhere that explains what this function is supposed to return? It accepts a list of strings and returns a list of strings. Typically in Python a mapping is represented as a `dict()`.

Reply:
Looking at the tests, it seems like this function returns a filtered version of the list it receives. In which case it may be better to name it `get_allowed_scopes` rather than `get_scopes_mapping` imo.